### PR TITLE
Update daily ci to use new cluster

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

And re-enable daily (which was disabled in #33136)

I haven't had a final confirmation from @glegendre01 , but let's take a chance to run it during this weekend.

### CODE REVIEW

File: .github/workflows/model_jobs.yml [hunk 1]
```diff
@@ -97,25 +98,42 @@ jobs:
...
+          if [ "${{ inputs.machine_type }}" = "aws-g4dn-2xlarge-cache" ]; then
+            machine_type=single-gpu
+          elif [ "${{ inputs.machine_type }}" = "aws-g4dn-12xlarge-cache" ]; then
+            machine_type=multi-gpu
+          else
+            machine_type=${{ inputs.machine_type }}
+          fi
+
+          echo "$machine_type"
+          echo "machine_type=$machine_type" >> $GITHUB_ENV
```
Comment (ydshieh) [inline .github/workflows/model_jobs.yml:116]: The new cluster is named `aws-g4dn-2xlarge-cache` or `aws-g4dn-12xlarge-cache`, but we need `single-gpu` and/or `multi-gpu` for the report file name or artifact name.

File: .github/workflows/self-scheduled-caller.yml [hunk 2]
```diff
@@ -2,6 +2,9 @@ name: Self-hosted runner (scheduled)
 
 
 on:
+  repository_dispatch:
+  schedule:
+    - cron: "17 2 * * *"
```
Comment (ydshieh) [inline .github/workflows/self-scheduled-caller.yml:7]: re-enable

File: .github/workflows/self-scheduled.yml [hunk 3]
```diff
@@ -102,7 +103,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        machine_type: [single-gpu, multi-gpu]
+        machine_type: [aws-g4dn-2xlarge-cache, aws-g4dn-12xlarge-cache]
```
Comment (ydshieh) [inline .github/workflows/self-scheduled.yml:106]: new cluster

**Review by amyeroberts — APPROVED** [review]
Thanks for updating! 

Only comment is that the mapping logic from the aws cluster name to "x-gpu" is very repetitive. I don't know of an obvious way to abstract this out in the yaml, but it would be cool if we could :)

**Comment by ydshieh** [comment]
> Thanks for updating!
> 
> Only comment is that the mapping logic from the aws cluster name to "x-gpu" is very repetitive. I don't know of an obvious way to abstract this out in the yaml, but it would be cool if we could :)

Yes I agree. I will think of it but let's try to run the CI during this weekend and potentially give feedbacks to the infra team on Monday 🙏 Thank you for the review.

**Pull request merged**